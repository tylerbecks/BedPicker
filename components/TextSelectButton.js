import React from 'react';
import { StyleSheet, Text } from 'react-native';
import CircleButton from './CircleButton';

const TextSelectButton = ({ text, onPress, selected }) => (
  <CircleButton onPress={onPress} style={selected ? styles.button : {}}>
    <Text
      style={
        selected ? { ...styles.text, ...styles.selectedText } : styles.text
      }
    >
      {text}
    </Text>
  </CircleButton>
);

export default TextSelectButton;

const styles = StyleSheet.create({
  text: {
    color: 'black',
    fontWeight: 'bold',
    textAlign: 'center'
  },
  selectedText: {
    color: 'white'
  },
  button: {
    backgroundColor: '#2196F3'
  }
});
