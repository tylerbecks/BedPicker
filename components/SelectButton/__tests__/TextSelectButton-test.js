import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import TextSelectButton from "../TextSelectButton";

it("renders", () => {
  expect(
    renderer
      .create(<TextSelectButton text="foo" onPress={jest.fn()} selected />)
      .toJSON()
  ).toMatchSnapshot();
});
