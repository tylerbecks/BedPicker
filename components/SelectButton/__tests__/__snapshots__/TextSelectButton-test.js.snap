// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2196F3",
      "borderRadius": 90,
      "elevation": 2,
      "height": 90,
      "justifyContent": "center",
      "margin": 10,
      "opacity": 1,
      "shadowColor": "rgba(0,0,0, .4)",
      "shadowOffset": Object {
        "height": 1,
        "width": 1,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
      "width": 90,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    foo
  </Text>
</View>
`;
